components/capsule.tsx

```diff
@@ -1,32 +1,35 @@
 import { ReactElement, ReactNode, useEffect, useRef, useState } from "react";
 import { motion } from "framer-motion";
 import styled, { css, CSSProp } from "styled-components";
+import { RemixiconComponentType } from "@remixicon/react";
 
 export interface CapsuleContentProps {
   id: string;
-  title: string;
+  title?: string;
   content?: ReactNode;
+  icon?: RemixiconComponentType | null;
 }
 
 export interface CapsuleProps {
   activeTab: string | null;
   tabs: CapsuleContentProps[];
-  setActiveTab?: (data: string) => void;
+  onTabChange?: (data: string) => void;
```

Comment:
let's rename the arg from `data` to `id` ... as much as possible let's be precise :)